podcastfy/client_v2.py

```diff
@@ -0,0 +1,239 @@
+import os
+import uuid
+import typer
+from pathlib import Path
+from typing import List, Optional, Dict, Any, Union
+
+from podcastfy.aiengines.llm.legacy_gemini_langchain import DefaultPodcastifyTranscriptEngine
+from podcastfy.aiengines.tts.tts_backends import OpenAITTS, ElevenLabsTTS, EdgeTTS
+from podcastfy.core.character import Character
+from podcastfy.core.podcast import Podcast, SyncTTSBackend, AsyncTTSBackend
+from podcastfy.core.transcript import Transcript
+from podcastfy.content_parser.content_extractor import ContentExtractor
+from podcastfy.core.tts_configs import TTSConfig
+from podcastfy.utils.config import Config, load_config
+from podcastfy.utils.logger import setup_logger
+
+logger = setup_logger(__name__)
+
+app = typer.Typer()
+
+def create_characters(config: Dict[str, Any]) -> List[Character]:
+    host = Character(
```

Comment:
I don't think the Character should be 'aware' of TTS or LLM. It's the opposite way around.

We should simply define Character's characteristics as you did, e.g. name, role as well as conversation_style, accent etc and then when we instantiate the LLM we use that info in its template etc

Reply:
That makes total sense, but in the same time I think it's great to have characters that have the same voice and settings independently from global settings on the tts. I don't know tts systems very well, but I think they strongly differ in their abilities to parameterize voices. But if one tts is highly flexible, I think the character is the right place to leverage this flexibility, rather than in the global settings. 

Your idea of having more tts neutral attributes in the instance do make sense to me, but that means the tts would have to translate this into parameters it can use, with some loss of expressiveness I think, or ignore altogether in the case of openai for example.

What about making it clear that it's optional like using words like `optional_affinities` ? To me that would be the best of both worlds. That would hold voice, model and extra_tts_args I think.